Coredump: switch format to Wasm module (#197)
WebAssembly/tool-conventions: Coredump.md

# Coredump file format

The coredump file is encoded using the [Wasm binary format], containing:
- general information about the process.
- the threads and stack frames.
- a snapshot of the WebAssembly linear memory or relevant regions.

The order in which they appear in the file is not important.

## Process information

> General informations about the process are stored in a [Custom section], called

Reviewer: ```suggestion
General information about the process is stored in a [Custom section], called
```

`core`.

```

> core         ::= customsec(procces-info)

Reviewer: ```suggestion
core         ::= customsec(process-info)
```

process-info ::= 0x0 executable-name:name
```

## Thread(s) and stack frames

> For each threads a [Custom section], called `corestack`, is used to store the

Reviewer: ```suggestion
For each thread a [Custom section], called `corestack`, is used to store the
```

debugging information.

```
corestack  ::= customsec(thread-info vec(frame))
thread-info ::= 0x0 thread-name:name
frame       ::= codeoffset:u32
                locals:vec(u32)
                stack:vec(u32)
                reserved:u32
```

The `reserved` byte is decoded as an empty vector and reserved for future use.

## Memory

The process memory is captured in the [Data Section], as active data segment(s).

## Limits

> - `executable-name`: limited to 16 characters

Reviewer: any particular reason for this limitation?

Author: I looked at a Google implementation of a user-space coredump (https://github.com/jeremyong/google-coredumper/blob/master/src/elfcore.c#L201) which has this 16 characters limitation. However, doing more research, I can't find any mention of this limitation. I'll remove it.
